crypto/init.c

```diff
@@ -130,7 +149,7 @@ DEFINE_RUN_ONCE_STATIC(ossl_init_base)
          */
 #  endif
         DSO_free(dso);
-        ERR_pop_to_mark();
+        err_unshelve_state(err);
```

Comment:
This does not seems to do the same as the old code and looks wrong to me.

Reply:
I misunderstood things, and it's probably fine.

Reply:
yes, while ERR_pop_to_mark does not always restore the error state (i.e. overflow),
err_unshelve_state is actually guaranteed to restore the exact state.